### PR TITLE
[release-v3.27] Auto pick #8085: Allow custom cgroup2 path

### PR DESCRIPTION
**Pull request by tomastigera**
Cherry pick of #8085 on release-v3.27.

#8085: Allow custom cgroup2 path

# Original PR Body below

## Description
Currently, the mount-bpffs init container tries to create the folder /run/calico/cgroup for mounting the cgroupv2 fs

This causes issues on distros like Talos Linux as described in the issue https://github.com/projectcalico/calico/issues/7892

This can be a configurable path instead

## Related issues/PRs
https://github.com/projectcalico/calico/issues/7892

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: alternative cgroup2 mount path can be specified by setting CALICO_CGROUP_PATH evn var for node.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Review by mazdakn — APPROVED** [review]
LGTM

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Comment by tomastigera** [comment]
unrelated failures in CI

**Pull request merged**
